src/app/public/modules/link/link-external.directive.ts

```diff
@@ -27,39 +32,55 @@ import { SkyAppLinkQueryParams } from './link-query-params';
 @Directive({
   selector: '[skyAppLinkExternal]'
 })
-export class SkyAppLinkExternalDirective extends RouterLinkWithHref {
-
-  private _queryParams: SkyAppLinkQueryParams;
+export class SkyAppLinkExternalDirective extends RouterLinkWithHref implements OnChanges {
 
   @Input()
   set skyAppLinkExternal(commands: any[] | string) {
     this.routerLink = commands;
   }
 
-  @Input()
-  set queryParams(params: SkyAppLinkQueryParams) {
-    this._queryParams = Object.assign(params, this.skyAppConfig.runtime.params.getAll());
-  }
-
-  get queryParams(): SkyAppLinkQueryParams {
-    if (!this._queryParams) {
-      this._queryParams = Object.assign({}, this.skyAppConfig.runtime.params.getAll());
-    }
-    return this._queryParams;
-  }
-
   constructor(
     router: Router,
     route: ActivatedRoute,
     platformLocation: PlatformLocation,
-    private skyAppConfig: SkyAppConfig,
-    private window: SkyAppWindowRef
+    private window: SkyAppWindowRef,
+    @Optional() private skyAppConfig?: SkyAppConfig,
+    @Optional() private paramsProvider?: SkyAppRuntimeConfigParamsProvider,
+    @Optional() hostConfig?: SkyAppConfigHost
   ) {
-    super(router, route, new PathLocationStrategy(platformLocation, skyAppConfig.skyux.host.url));
+    super(
+      router,
+      route,
+      new PathLocationStrategy(
+        platformLocation,
+        (skyAppConfig)
```

Comment:
Assuming we want to eventually deprecate `SkyAppConfig` and use `SkyAppConfigHost` instead, should we default to `SkyAppConfigHost` and fall back to `SkyAppConfig`?